bundles/org.openhab.binding.paradoxalarm/src/main/java/org/openhab/binding/paradoxalarm/internal/handlers/ParadoxIP150BridgeHandler.java

```diff
@@ -0,0 +1,276 @@
+
+            boolean isOnline = communicator.isOnline();
+            cache.setIsOnline(isOnline);
+
+            if (isOnline) {
+                communicator.refreshMemoryMap();
+
+                cache.setPanelInfoBytes(communicator.getPanelInfoBytes());
+                cache.setPartitionStateFlags(communicator.readPartitionFlags());
+                cache.setZoneStateFlags(communicator.readZoneStateFlags());
+
+                if (withEpromValues) {
+                    cache.setPartitionLabels(communicator.readPartitionLabels());
+                    cache.setZoneLabels(communicator.readZoneLabels());
+                }
+            }
+
+            announceUpdateToHandlers();
+
+            if (failedReadAttempts > 0) {
+                logger.info("Successfully refreshed memory map after {} failed attempts.", failedReadAttempts);
+                failedReadAttempts = 0;
+            }
+
+        } catch (IOException e) {
+            handleSocketReadError(e);
+        } catch (InterruptedException | ParadoxBindingException e) {
+            logger.warn("Exception while refreshing memory map in communicator.", e);
+        }
+    }
+
+    private void handleSocketReadError(IOException e) {
+        failedReadAttempts++;
+        logger.warn("Communicator cannot refresh cached memory map. Attempt: {}, IOException msg: {}",
+                failedReadAttempts, e.getMessage());
+        logger.debug("Stack trace:", e);
+
+        if (failedReadAttempts == FAILED_READ_ATTEMPTS_TRESHOLD) {
+            logger.info("Will attempt to reinitialize communicator.");
+            ChannelUID uid = getThing().getChannel(ParadoxAlarmBindingConstants.IP150_COMMUNICATION_COMMAND_CHANNEL_UID)
+                    .getUID();
+            handleCommand(uid, new StringType(RESET_COMMAND));
+        }
+    }
+
+    private void announceUpdateToHandlers() {
+        Bridge bridge = getThing();
+        List<Thing> things = bridge.getThings();
+        for (Thing thing : things) {
+            ThingHandler handler = thing.getHandler();
+            Channel bridgeChannel = bridge
+                    .getChannel(ParadoxAlarmBindingConstants.IP150_COMMUNICATION_COMMAND_CHANNEL_UID);
+            if (handler != null && bridgeChannel != null) {
+                handler.handleCommand(bridgeChannel.getUID(), RefreshType.REFRESH);
+            }
+        }
+    }
+
+    public IParadoxCommunicator resetCommunicator()
+            throws UnknownHostException, IOException, InterruptedException, ParadoxBindingException {
+        synchronized (communicator) {
+            if (communicator != null) {
+                communicator.close();
+            }
+            communicator = initializeCommunicator();
+
+            return communicator;
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Received command {}", command.toFullString());

```

Comment:
Do you really need the full string now you will always (calculate) the full string even if the logging does not apply